### PR TITLE
default relationship can now refer to different namespace

### PR DESCRIPTION
**Pull request by aaronAB1993**
#227

### CODE REVIEW

File: osp/core/ontology/yml/yml_parser.py [hunk 1]
```diff
@@ -404,8 +406,51 @@ def _set_inverse(self, entity_name, entity_doc):
...
+        a default relationship, save that accordingly.
+        """
+        if DEFAULT_REL_KEY in self._doc:
+            namespace, entity_name = self._doc[DEFAULT_REL_KEY].split('.')
+
+            # defined relationship must be installed
+            from osp.core.namespaces import _namespace_registry
+            referred_namespace = _namespace_registry.get(namespace)
+            if not referred_namespace:
+                raise ValueError(f"""The namespace {namespace} that you
```
Comment (urbanmatthias) [inline osp/core/ontology/yml/yml_parser.py:436]: I prefer using
```py
ValueError(f"line1 "
           f"line2 "
           f"line3 ")
```
over 
```py
ValueError(f"""line1
    line2
    line3""")
```
if the string does not fit in one line.
The first string results in `line1 line2 line3` while the latter results in

```
line1
    line2
    line3
```

Comment (urbanmatthias) [reply osp/core/ontology/yml/yml_parser.py:436]: ValueError: You have defined 2 default relationships for namespace default_rel_test_namespace_two_definitions although <= 1 are allowed.

vs.

ValueError: You have defined 2 default
        relationships for namespace default_rel_test_namespace_two_definitions
        although <= 1 are allowed.

Comment (aaronAB1993) [reply osp/core/ontology/yml/yml_parser.py:436]: Alright, thanks :)

File: osp/core/ontology/yml/yml_parser.py [hunk 2]
```diff
@@ -404,8 +406,51 @@ def _set_inverse(self, entity_name, entity_doc):
             (iri, rdflib.OWL.inverseOf, inverse_iri)
         )
 
-    def _check_default_rel(self, entity_name, entity_doc):
-        """Check if the given relationship the default
+    def _assert_default_relationship_occurance(self):
```
Comment (urbanmatthias) [inline osp/core/ontology/yml/yml_parser.py:409]: Occurrence is the correct way to spell this word, meaning 'an event that happened.'

Occurance is one of the most popular ways that this world is misspelt.

https://hinative.com/en-US/questions/1554753

Is this true?

Comment (urbanmatthias) [reply osp/core/ontology/yml/yml_parser.py:409]: I will update it to occurrence

Comment (aaronAB1993) [reply osp/core/ontology/yml/yml_parser.py:409]: True: https://www.dict.cc/?s=occurrence

But I am happy to be one out of millions :)

File: tests/test_namespace.py [hunk 3]
```diff
@@ -312,10 +313,37 @@ def test_namespace_str(self):
                          "<city: http://www.osp-core.com/city#>")
 
     def test_get_default_rel(self):
+        # default rel defined as flag in entity name
         self.installer.install("city")
         namespace = self.namespace_registry.city
         self.assertEqual(namespace.get_default_rel().name, "hasPart")
 
+        onto_def_rel = os.path.join(
```
Comment (urbanmatthias) [inline tests/test_namespace.py:321]: You're technically not testing the namespace or the get_default method, as the method name or the file name of this test might suggest.

But I guess that's fine. What do you think?

Comment (aaronAB1993) [reply tests/test_namespace.py:321]: I think you are right, there is no need to install the namespace, i can rather directly test the `YmlParser.parse()`. I will put the test somewhere else.

Comment (aaronAB1993) [reply tests/test_namespace.py:321]: Actually, I just found that it is ok. What I want to test is that the installation of the yml will result in the right default relationship in the namespace. I think the confusion is that the `YmlParser` is used for installation and this is where I put my code. So the question is, is this test part of the installation, the namespace or the parser? I could change the name of the test though.

Comment (urbanmatthias) [reply tests/test_namespace.py:321]: Let it as it is. I think it is fine

**Comment by urbanmatthias** [comment]
I think you can merge now

**Pull request merged**